gpkit/keydict.py

```diff
@@ -0,0 +1,229 @@
+"Implements KeyDict and KeySet classes"
+import numpy as np
+from collections import defaultdict
+from .small_classes import Numbers, Strings
+from .small_scripts import is_sweepvar, veckeyed
+
+
+class KeyDict(dict):
+    """KeyDicts allow storing and accessing the same value with multiple keys
+
+    A KeyDict keeps an internal list of VarKeys as canonical keys, but allows
+    accessing their values with any object whose `key` attribute matches
+    one of those VarKeys, or with strings who match any of the multiple
+    possible string interpretations of each key.
+
+    ```
+    kd = gpkit.keydict.KeyDict()
+    x = gpkit.Variable("x", model="test")
+    kd[x] = 1
+    assert kd[x] == kd[x.key] == kd["x"] == kd["x_test"] == 1
```

Comment:
We test its use (the functionality above) quite often, but not in tests that are labeled "KeyDict"